src/zeta/aca_zeta_programming.cpp

```diff
@@ -75,14 +75,16 @@ int ACA_Zeta_Programming::delete_zeta_config(const alcor::schema::AuxGateway cur
   for (auto destination : current_AuxGateway.destinations()) {
     stZetaCfg.zeta_buckets.push_back(destination.ip_address());
   }
-  uint32_t oam_server_port = current_AuxGateway.zeta_info().port_inband_operation();
+  uint oam_server_port = current_AuxGateway.zeta_info().port_inband_operation();
 
   // Reset oam_server_port to 0
-  ACA_Vlan_Manager::get_instance().set_oam_server_port(vpc_id, 0);
+  ACA_Vlan_Manager::get_instance().set_oam_server_port(tunnel_id, 0);
 
   // update oam_ports_table and delete the OAM punt rule if the last port in the VPC has been deleted
-  Aca_Oam_Port_Manager::get_instance().remove_vpc(oam_server_port, tunnel_id);
 
+  if (!ACA_Vlan_Manager::get_instance().is_exist_oam_port(oam_server_port)) {
+    Aca_Oam_Port_Manager::get_instance().remove_port_number(oam_server_port);
+  }
   // delete the group bucket rule
   overall_rc = _delete_zeta_group_entry(&stZetaCfg);
```

Comment:
line 87 is deleting the group rules for deleting:
1. second port in the VPC - not good as the first port still need to use it
2. first port in same VPC - not usable because the group rule has been deleted above

We need to fix the code so that only the last port using the ZGC will delete the group rule